terraform-provider-constellation/internal/provider/cluster_resource.go

```diff
@@ -0,0 +1,308 @@
+				Description:         "The Constellation OS image to use in the CSP specific reference format. Use the `constellation_image` data source to find the correct image for your CSP. When not set, the latest default version will be used.",
+				Optional:            true,
+			},
+			"kubernetes_version": schema.StringAttribute{
+				MarkdownDescription: "The Kubernetes version to use for the cluster.",
+				Description:         "The Kubernetes version to use for the cluster. When not set, the latest default version will be used.", // TODO(elchead): refer to supported versions; wait for https://github.com/edgelesssys/constellation/pull/2661
+				Optional:            true,
+			},
+			"constellation_microservices_version": schema.StringAttribute{
+				MarkdownDescription: "The Constellation microservices version to use for the cluster.",
+				Description:         "The Constellation microservices version to use for the cluster. When not set, the latest default version will be used.",
+				Optional:            true,
+			},
+			"debug": schema.BoolAttribute{
+				MarkdownDescription: "~> **Warning:** Do not enable Debug mode in production environments.\nEnable debug mode and allow the use of debug images.",
+				Description:         "DON'T USE IN PRODUCTION: Enable debug mode and allow the use of debug images.",
+				Optional:            true,
+			},
+			"init_endpoint": schema.StringAttribute{
+				MarkdownDescription: "The endpoint to use for cluster initialization. This is the endpoint of the node running the bootstrapper.",
+				Description:         "The endpoint to use for cluster initialization.",
+				Optional:            true,
+			},
+			"kubernetes_api_endpoint": schema.StringAttribute{
+				MarkdownDescription: "The endpoint to use for the Kubernetes API.",
+				Description:         "The endpoint to use for the Kubernetes API. When not set, the default endpoint will be used.",
+				Optional:            true,
+			},
+			"extra_microservices": schema.SingleNestedAttribute{
+				MarkdownDescription: "Extra microservice settings.",
+				Description:         "Extra microservice settings.",
+				Optional:            true,
+				Attributes: map[string]schema.Attribute{
+					"csi_driver": schema.BoolAttribute{
+						Optional:            true,
+						MarkdownDescription: "Enable the CSI driver microservice.",
+						Description:         "Enable the CSI driver microservice.",
+					},
+				},
+			},
+			"master_secret": schema.StringAttribute{
+				MarkdownDescription: "The master secret to use for the cluster.",
+				Description:         "The master secret to use for the cluster.",
+				Required:            true,
+			},
+			"init_secret": schema.StringAttribute{
+				MarkdownDescription: "The init secret to use for the cluster.",
+				Description:         "The init secret to use for the cluster.",
+				Required:            true,
+			}, // TODO merge / derive from master secret?
```

Comment:
We shouldn't merge this with the master secret, since within a VPC, the init secret is not considered to be secret (it is attached as a tag to the control plane nodes).
Deriving it from the master secret would require it to be generated by the user with a by us specify , so I don't think this is a good approach either.
I would just keep it as separate input